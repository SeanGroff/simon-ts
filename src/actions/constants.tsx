export const TOGGLE_GAME_POWER = 'TOGGLE_GAME_POWER';
export type TOGGLE_GAME_POWER = typeof TOGGLE_GAME_POWER;

export const START_GAME = 'START_GAME';
export type START_GAME = typeof START_GAME;

export const START_NEXT_ROUND = 'START_NEXT_ROUND';
export type START_NEXT_ROUND = typeof START_NEXT_ROUND;

export const ROUND_SUCCESS = 'ROUND_SUCCESS';
export type ROUND_SUCCESS = typeof ROUND_SUCCESS;

export const SET_LIGHT_SEQUENCE = 'SET_LIGHT_SEQUENCE';
export type SET_LIGHT_SEQUENCE = typeof SET_LIGHT_SEQUENCE;
