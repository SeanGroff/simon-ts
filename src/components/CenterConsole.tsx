import * as React from 'react';
import styled from 'styled-components';
import Counter from './Counter';
import StartStrictButton from './StartStrictButton';
import OnOffSwitch from './OnOffSwitch';

interface Props {
  hasMargin?: boolean;
  hasPadding?: boolean;
  power: boolean;
  counter: number;
  toggleGamePowerAction(power: boolean): { type: string; payload: boolean };
  startGameAction(): { type: string };
}

interface LabelProps {
  hasMargin?: boolean;
  hasPadding?: boolean;
}

const StyledWrapper = styled.div`
  position: absolute;
  border-radius: 100%;
  top: 50%;
  left: 50%;
  background-color: #ece7ee;
  border: 12px solid #333333;
  width: 220px;
  height: 220px;
  margin: -120px 0 0 -120px;
`;

const TopHalf = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #333333;
  width: 220px;
  height: 110px;
  border-top-left-radius: 220px;
  border-top-right-radius: 220px;
  margin-top: -13px;
`;

const BottomHalf = TopHalf.extend`
  flex-direction: column;
  background-color: #ece7ee;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 220px;
  border-bottom-right-radius: 220px;
  margin-top: 0;
`;

const Header = styled.span`
  color: #ece7ee;
  font-family: 'Anton', sans-serif;
  font-size: 52px;
  letter-spacing: 2px;
  margin-top: 35px;
  text-transform: capitalize;
`;

const ButtonsRow = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-around;
`;

const ButtonWrapper = styled.span`
  display: flex;
  flex-direction: column;
  margin: 5px;
`;

const StyledLabel = styled.label`
  font-size: 14px;
  color: #222222;
  margin: ${({ hasMargin }: LabelProps) =>
    hasMargin ? 'auto 5px 0 5px' : '0'};
  padding: ${({ hasPadding }: LabelProps) => (hasPadding ? '0 0 5px 0' : '0')};
  text-transform: capitalize;
  text-align: center;
`;

export default (props: Props) => (
  <StyledWrapper>
    <TopHalf>
      <Header>{'simon'}</Header>
    </TopHalf>
    <BottomHalf>
      <ButtonsRow>
        <ButtonWrapper>
          <StyledLabel hasPadding={true}>{'count'}</StyledLabel>
          <Counter count={props.counter} on={props.power} />
        </ButtonWrapper>
        <ButtonWrapper>
          <StyledLabel hasPadding={true}>{'start'}</StyledLabel>
          <StartStrictButton
            startButton={true}
            disabled={!props.power || props.counter ? true : false}
            power={props.power}
            startGame={props.startGameAction}
          />
        </ButtonWrapper>
        <ButtonWrapper>
          <StyledLabel hasPadding={true}>{'strict'}</StyledLabel>
          <StartStrictButton
            startButton={false}
            disabled={!props.power}
            power={props.power}
          />
        </ButtonWrapper>
      </ButtonsRow>
      <ButtonsRow>
        <StyledLabel hasMargin={true}>{'off'}</StyledLabel>
        <OnOffSwitch
          power={props.power}
          powerOnOff={props.toggleGamePowerAction}
        />
        <StyledLabel hasMargin={true}>{'on'}</StyledLabel>
      </ButtonsRow>
    </BottomHalf>
  </StyledWrapper>
);
