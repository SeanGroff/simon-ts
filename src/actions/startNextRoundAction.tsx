import * as types from '../types/types';
import * as constants from './constants';

type Action = types.StartNextRoundAction;

export const startNextRoundAction = (): Action => ({
  type: constants.START_NEXT_ROUND,
});

// type Args = {
//   playerTurn: number,
//   counter: number,
// };

/**
 * @todo Write test for this Thunk
 *
 * @export
 * @returns
 */
// export function nextTurnThunk() {
//   return (dispatch: *, getState: *) => {
//     const { playerTurn, counter }: Args = getState();
//     setTimeout(() => {
//       dispatch(startNextRoundAction(!playerTurn));
//     }, counter * 1000);
//   };
// }
