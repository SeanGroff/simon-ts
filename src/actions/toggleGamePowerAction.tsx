import * as constants from './constants';
import * as types from '../types/types';

type Action = types.ToggleGamePowerAction;

export const toggleGamePowerAction = (payload: boolean): Action => ({
  type: constants.TOGGLE_GAME_POWER,
  payload,
});
