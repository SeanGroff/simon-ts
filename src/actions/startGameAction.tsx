import { Dispatch } from 'redux';
import * as constants from './constants';
import * as types from '../types/types';
import { startNextRoundAction as startNextRound } from './startNextRoundAction';

export const startGameAction = (): types.StartGameAction => ({
  type: constants.START_GAME,
});

/**
 * @todo Finish startGameThunk (See comments)
 */
export function startGameThunk(): types.Thunk {
  return (dispatch: Dispatch<types.StoreState>) => {
    // const { playerTurn, counter }: Args = getState();

    dispatch(startGameAction());
    dispatch(startNextRound());
    //  dispatch(lightSequence()) // each interval setState to color, playerTurn = true
  };
}
