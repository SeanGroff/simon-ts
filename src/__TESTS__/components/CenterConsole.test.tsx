global.requestAnimationFrame = callback => {
  setTimeout(callback, 0);
};

import * as React from 'react';
import { shallow, configure } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import CenterConsole from '../../components/CenterConsole';
import { toggleGamePowerAction } from '../../actions/toggleGamePowerAction';
import { startGameThunk } from '../../actions/startGameAction';

it('renders correctly', () => {
  const component = shallow(
    <CenterConsole
      power={true}
      counter={0}
      toggleGamePowerAction={toggleGamePowerAction}
      startGameThunk={startGameThunk}
    />,
  );
  expect(component).toMatchSnapshot();
});
