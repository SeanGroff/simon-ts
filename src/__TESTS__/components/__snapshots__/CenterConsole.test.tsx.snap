// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    counter={0}
    power={true}
    startGameThunk={[Function]}
    toggleGamePowerAction={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.div>
          <styled.span>
                    simon
          </styled.span>
</styled.div>,
        <styled.div>
          <styled.div>
                    <styled.span>
                              <styled.label
                                        hasPadding={true}
                              >
                                        count
                              </styled.label>
                              <Unknown
                                        count={0}
                                        on={true}
                              />
                    </styled.span>
                    <styled.span>
                              <styled.label
                                        hasPadding={true}
                              >
                                        start
                              </styled.label>
                              <StartStrictButton
                                        disabled={false}
                                        power={true}
                                        startButton={true}
                                        startGame={[Function]}
                              />
                    </styled.span>
                    <styled.span>
                              <styled.label
                                        hasPadding={true}
                              >
                                        strict
                              </styled.label>
                              <StartStrictButton
                                        disabled={false}
                                        power={true}
                                        startButton={false}
                              />
                    </styled.span>
          </styled.div>
          <styled.div>
                    <styled.label
                              hasMargin={true}
                    >
                              off
                    </styled.label>
                    <OnOffSwitch
                              power={true}
                              powerOnOff={[Function]}
                    />
                    <styled.label
                              hasMargin={true}
                    >
                              on
                    </styled.label>
          </styled.div>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": <styled.span>
            simon
</styled.span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": "simon",
          },
          "ref": null,
          "rendered": "simon",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.div>
              <styled.span>
                            <styled.label
                                          hasPadding={true}
                            >
                                          count
                            </styled.label>
                            <Unknown
                                          count={0}
                                          on={true}
                            />
              </styled.span>
              <styled.span>
                            <styled.label
                                          hasPadding={true}
                            >
                                          start
                            </styled.label>
                            <StartStrictButton
                                          disabled={false}
                                          power={true}
                                          startButton={true}
                                          startGame={[Function]}
                            />
              </styled.span>
              <styled.span>
                            <styled.label
                                          hasPadding={true}
                            >
                                          strict
                            </styled.label>
                            <StartStrictButton
                                          disabled={false}
                                          power={true}
                                          startButton={false}
                            />
              </styled.span>
</styled.div>,
            <styled.div>
              <styled.label
                            hasMargin={true}
              >
                            off
              </styled.label>
              <OnOffSwitch
                            power={true}
                            powerOnOff={[Function]}
              />
              <styled.label
                            hasMargin={true}
              >
                            on
              </styled.label>
</styled.div>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.span>
                  <styled.label
                                    hasPadding={true}
                  >
                                    count
                  </styled.label>
                  <Unknown
                                    count={0}
                                    on={true}
                  />
</styled.span>,
                <styled.span>
                  <styled.label
                                    hasPadding={true}
                  >
                                    start
                  </styled.label>
                  <StartStrictButton
                                    disabled={false}
                                    power={true}
                                    startButton={true}
                                    startGame={[Function]}
                  />
</styled.span>,
                <styled.span>
                  <styled.label
                                    hasPadding={true}
                  >
                                    strict
                  </styled.label>
                  <StartStrictButton
                                    disabled={false}
                                    power={true}
                                    startButton={false}
                  />
</styled.span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.label
                      hasPadding={true}
>
                      count
</styled.label>,
                    <Unknown
                      count={0}
                      on={true}
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "count",
                      "hasPadding": true,
                    },
                    "ref": null,
                    "rendered": "count",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "count": 0,
                      "on": true,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.label
                      hasPadding={true}
>
                      start
</styled.label>,
                    <StartStrictButton
                      disabled={false}
                      power={true}
                      startButton={true}
                      startGame={[Function]}
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "start",
                      "hasPadding": true,
                    },
                    "ref": null,
                    "rendered": "start",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "disabled": false,
                      "power": true,
                      "startButton": true,
                      "startGame": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <styled.label
                      hasPadding={true}
>
                      strict
</styled.label>,
                    <StartStrictButton
                      disabled={false}
                      power={true}
                      startButton={false}
/>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "class",
                    "props": Object {
                      "children": "strict",
                      "hasPadding": true,
                    },
                    "ref": null,
                    "rendered": "strict",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "function",
                    "props": Object {
                      "disabled": false,
                      "power": true,
                      "startButton": false,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <styled.label
                  hasMargin={true}
>
                  off
</styled.label>,
                <OnOffSwitch
                  power={true}
                  powerOnOff={[Function]}
/>,
                <styled.label
                  hasMargin={true}
>
                  on
</styled.label>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "off",
                  "hasMargin": true,
                },
                "ref": null,
                "rendered": "off",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "power": true,
                  "powerOnOff": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": "on",
                  "hasMargin": true,
                },
                "ref": null,
                "rendered": "on",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.div>
            <styled.span>
                        simon
            </styled.span>
</styled.div>,
          <styled.div>
            <styled.div>
                        <styled.span>
                                    <styled.label
                                                hasPadding={true}
                                    >
                                                count
                                    </styled.label>
                                    <Unknown
                                                count={0}
                                                on={true}
                                    />
                        </styled.span>
                        <styled.span>
                                    <styled.label
                                                hasPadding={true}
                                    >
                                                start
                                    </styled.label>
                                    <StartStrictButton
                                                disabled={false}
                                                power={true}
                                                startButton={true}
                                                startGame={[Function]}
                                    />
                        </styled.span>
                        <styled.span>
                                    <styled.label
                                                hasPadding={true}
                                    >
                                                strict
                                    </styled.label>
                                    <StartStrictButton
                                                disabled={false}
                                                power={true}
                                                startButton={false}
                                    />
                        </styled.span>
            </styled.div>
            <styled.div>
                        <styled.label
                                    hasMargin={true}
                        >
                                    off
                        </styled.label>
                        <OnOffSwitch
                                    power={true}
                                    powerOnOff={[Function]}
                        />
                        <styled.label
                                    hasMargin={true}
                        >
                                    on
                        </styled.label>
            </styled.div>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": <styled.span>
              simon
</styled.span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": "simon",
            },
            "ref": null,
            "rendered": "simon",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.div>
                <styled.span>
                                <styled.label
                                                hasPadding={true}
                                >
                                                count
                                </styled.label>
                                <Unknown
                                                count={0}
                                                on={true}
                                />
                </styled.span>
                <styled.span>
                                <styled.label
                                                hasPadding={true}
                                >
                                                start
                                </styled.label>
                                <StartStrictButton
                                                disabled={false}
                                                power={true}
                                                startButton={true}
                                                startGame={[Function]}
                                />
                </styled.span>
                <styled.span>
                                <styled.label
                                                hasPadding={true}
                                >
                                                strict
                                </styled.label>
                                <StartStrictButton
                                                disabled={false}
                                                power={true}
                                                startButton={false}
                                />
                </styled.span>
</styled.div>,
              <styled.div>
                <styled.label
                                hasMargin={true}
                >
                                off
                </styled.label>
                <OnOffSwitch
                                power={true}
                                powerOnOff={[Function]}
                />
                <styled.label
                                hasMargin={true}
                >
                                on
                </styled.label>
</styled.div>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.span>
                    <styled.label
                                        hasPadding={true}
                    >
                                        count
                    </styled.label>
                    <Unknown
                                        count={0}
                                        on={true}
                    />
</styled.span>,
                  <styled.span>
                    <styled.label
                                        hasPadding={true}
                    >
                                        start
                    </styled.label>
                    <StartStrictButton
                                        disabled={false}
                                        power={true}
                                        startButton={true}
                                        startGame={[Function]}
                    />
</styled.span>,
                  <styled.span>
                    <styled.label
                                        hasPadding={true}
                    >
                                        strict
                    </styled.label>
                    <StartStrictButton
                                        disabled={false}
                                        power={true}
                                        startButton={false}
                    />
</styled.span>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.label
                        hasPadding={true}
>
                        count
</styled.label>,
                      <Unknown
                        count={0}
                        on={true}
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": "count",
                        "hasPadding": true,
                      },
                      "ref": null,
                      "rendered": "count",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "count": 0,
                        "on": true,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.label
                        hasPadding={true}
>
                        start
</styled.label>,
                      <StartStrictButton
                        disabled={false}
                        power={true}
                        startButton={true}
                        startGame={[Function]}
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": "start",
                        "hasPadding": true,
                      },
                      "ref": null,
                      "rendered": "start",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "disabled": false,
                        "power": true,
                        "startButton": true,
                        "startGame": [Function],
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": Array [
                      <styled.label
                        hasPadding={true}
>
                        strict
</styled.label>,
                      <StartStrictButton
                        disabled={false}
                        power={true}
                        startButton={false}
/>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "class",
                      "props": Object {
                        "children": "strict",
                        "hasPadding": true,
                      },
                      "ref": null,
                      "rendered": "strict",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": null,
                      "nodeType": "function",
                      "props": Object {
                        "disabled": false,
                        "power": true,
                        "startButton": false,
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <styled.label
                    hasMargin={true}
>
                    off
</styled.label>,
                  <OnOffSwitch
                    power={true}
                    powerOnOff={[Function]}
/>,
                  <styled.label
                    hasMargin={true}
>
                    on
</styled.label>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "off",
                    "hasMargin": true,
                  },
                  "ref": null,
                  "rendered": "off",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "power": true,
                    "powerOnOff": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": "on",
                    "hasMargin": true,
                  },
                  "ref": null,
                  "rendered": "on",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
